### PR TITLE
Feature/android start rework

### PR DESCRIPTION
**Pull request by arnaudgiuliani**
From issue https://github.com/InsertKoinIO/koin/issues/156, I propose to rework the startKoin for Android like follow: replace the need of `Application` to `Context` instead.

startKoin() can be use in mroe Android classes.

### CODE REVIEW

File: koin-projects/examples/android-mvvm-coroutines/src/main/kotlin/fr/ekito/myweatherapp/data/datasource/webservice/local/AndroidJsonReader.kt [hunk 1]
```diff
@@ -1,25 +1,24 @@
...
-import fr.ekito.myweatherapp.data.datasource.webservice.local.BaseReader
+import android.content.Context
 import java.io.BufferedReader
 import java.io.InputStreamReader
 
 /**
  * Read Json File from assets/json
  */
-class AndroidJsonReader(val application: Application) : BaseReader() {
+class AndroidJsonReader(val application: Context) : BaseReader() {
```
Comment (Baresse) [inline koin-projects/examples/android-mvvm-coroutines/src/main/kotlin/fr/ekito/myweatherapp/data/datasource/webservice/local/AndroidJsonReader.kt]: naming should be context and not application

File: koin-projects/examples/android-mvvm/src/main/kotlin/fr/ekito/myweatherapp/data/datasource/webservice/local/AndroidJsonReader.kt [hunk 2]
```diff
@@ -1,25 +1,24 @@
...
-import fr.ekito.myweatherapp.data.datasource.webservice.local.BaseReader
+import android.content.Context
 import java.io.BufferedReader
 import java.io.InputStreamReader
 
 /**
  * Read Json File from assets/json
  */
-class AndroidJsonReader(val application: Application) : BaseReader() {
+class AndroidJsonReader(val application: Context) : BaseReader() {
```
Comment (Baresse) [inline koin-projects/examples/android-mvvm/src/main/kotlin/fr/ekito/myweatherapp/data/datasource/webservice/local/AndroidJsonReader.kt]: naming should be context and not application

File: koin-projects/koin-android/src/docs/asciidoc/starting-with-android.adoc [hunk 3]
```diff
@@ -21,22 +21,22 @@ class MainApplication : Application() {
...
-If you need to start Koin from elsewhere, you can just use the static `statrKoin()` function and provide your Android `Application`
-instance with the `with` operator:
+If you need to start Koin from another Android class, you can use the `startKoin()` function and provide your Android `Context`
+instance with just like:
 
 [source,kotlin]
 ----
 // Start Koin and init Application instance definition
-startKoin(myAppModules) with (application)
+startKoin(androidContextn, myAppModules)
```
Comment (Baresse) [inline koin-projects/koin-android/src/docs/asciidoc/starting-with-android.adoc]: typo here `androidContextn` should be `androidContext`

File: koin-projects/koin-android/src/main/java/org/koin/android/ext/android/ApplicationExt.kt [hunk 4]
```diff
@@ -1,89 +0,0 @@
-/*
```
Comment (Baresse) [inline koin-projects/koin-android/src/main/java/org/koin/android/ext/android/ApplicationExt.kt]: Why did you removed this ? What about backward compat ? Is it mandatory ?

File: koin-projects/examples/android-mvvm-coroutines/src/main/kotlin/fr/ekito/myweatherapp/data/datasource/webservice/local/AndroidJsonReader.kt (hunk 1 above)
Comment (arnaudgiuliani) [reply koin-projects/examples/android-mvvm-coroutines/src/main/kotlin/fr/ekito/myweatherapp/data/datasource/webservice/local/AndroidJsonReader.kt]: yep

File: koin-projects/koin-android/src/main/java/org/koin/android/ext/android/ApplicationExt.kt (hunk 4 above)
Comment (arnaudgiuliani) [reply koin-projects/koin-android/src/main/java/org/koin/android/ext/android/ApplicationExt.kt]: I've upgrade the extension to ComponentCallbacks.statrKoin(), which is equivalent to old Application.statrKoin(). But it allows to use this startKoin from another Android class if needed (ContentProvider for example)

**Pull request merged**